File: ios/Sources/YoutubeDownloadPlugin/YoutubeDownloadPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(YoutubeDownloadPlugin)
public class YoutubeDownloadPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "YoutubeDownloadPlugin"
    public let jsName = "YoutubeDownload"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = YoutubeDownload()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }

    @objc func openMap(_ call: CAPPluginCall) {
        let latitude = call.getString("latitude")
        let longitude = call.getNumber("longitude")

        // more logic

        call.resolve()
    }
}

File: android/src/main/java/com/capacitor/plugins/youtube/download/YoutubeDownloadPlugin.java
package com.capacitor.plugins.youtube.download;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "YoutubeDownload")
public class YoutubeDownloadPlugin extends Plugin {

    private YoutubeDownload implementation = new YoutubeDownload();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

    @PluginMethod()
    public void openMap(PluginCall call) {
        Double latitude = call.getDouble("latitude");
        Double longitude = call.getDouble("longitude");

        // more logic

        call.resolve();
    }
}
